### PR TITLE
Adding optional custom DateHeader component for month view

### PR DESCRIPTION
**Pull request by andresgutgon**
Trying to solve this issue https://github.com/intljusticemission/react-big-calendar/issues/224

### CODE REVIEW

File: src/Calendar.js [hunk 1]
```diff
@@ -412,14 +411,17 @@ class Calendar extends React.Component {
 
      day: PropTypes.shape({
        header: elementType,
+       dateHeader: elementType,
        event: elementType
      }),
      week: PropTypes.shape({
        header: elementType,
+       dateHeader: elementType,
```
Comment (jquense) [inline src/Calendar.js]: This isn't relevant for Day or Week views

File: src/Month.js [hunk 2]
```diff
@@ -86,6 +87,7 @@ class MonthView extends React.Component {
       rowLimit: 5,
       needLimitMeasure: true,
     }
+    this.handleHeadingClick = this.handleHeadingClick.bind(this)
```
Comment (jquense) [inline src/Month.js]: Not necessary since you close over with an arrow function below

File: src/Month.js [hunk 3]
```diff
@@ -215,16 +218,10 @@ class MonthView extends React.Component {
...
-            >
-              {label}
-            </a>
-          : <span>
-              {label}
-            </span>}
+        <DateHeaderComponent
+          label={label}
+          drilldownView={drilldownView}
+          onDateClick={e => this.handleHeadingClick(date, drilldownView, e)} />
```
Comment (jquense) [inline src/Month.js]: Maybe onDrilldown

File: examples/demos/customDateHeader.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import BigCalendar from 'react-big-calendar';
+import events from '../events';
```
Comment (jquense) [inline examples/demos/customDateHeader.js]: No need for an example, but please add a story that demonstrates the functionality (in /stories)

**Comment by andresgutgon** [comment]
Thanks for reviewing! I'll address your comments

**Comment by andresgutgon** [comment]
Comments fixed :)

**Comment by andresgutgon** [comment]
### Offtopic:
I saw double `less` command here: https://github.com/intljusticemission/react-big-calendar/blob/master/package.json#L32

It's ok?

File: src/BackgroundCells.js [hunk 5]
```diff
@@ -62,7 +62,12 @@ class BackgroundCells extends React.Component {
         {range.map((date, index) => {
           let selected =  selecting && index >= startIdx && index <= endIdx;
           return (
-            <Wrapper key={index} value={date}>
+            <Wrapper
+              key={index}
+              value={date}
+              isFirstInRange={index === 0}
+              isLastInRange={index === range.length - 1}
```
Comment (jquense) [inline src/BackgroundCells.js]: ? I'd prefer to not pass arbitrary props to this component

Comment (andresgutgon) [inline src/BackgroundCells.js]: There is a reason for it. Maybe you can give me an alternative.
On our calendar when use clicks on a date we open a popover. If the day is the first one in the row popover open with arrow left aligned
image

If it's `last` cell in the row it opens in this way:
image


If it's not `first` or `last` it opens centered
image

Comment (andresgutgon) [inline src/BackgroundCells.js]: @jquense please let me know if make sense what I said. Maybe there is an easier way of do what we want.
Thanks!

Comment (jquense) [inline src/BackgroundCells.js]: it makes sense, tho its a bit specific to your needs ;) maybe just pass the range and value in and you can figure out if it's the first or not in your app code? I don't like having props that have no purpose in the code itself, it'll be really easy to break your usecase later without realizing

Comment (andresgutgon) [inline src/BackgroundCells.js]: deal

**Comment by andresgutgon** [comment]
@jquense done all changes 🎉

**Comment by andresgutgon** [comment]
@jquense please let me know if I need to change something else. I would like to have this code merged. Thanks 🙏

**Comment by jquense** [comment]
thanks!

**Comment by andresgutgon** [comment]
Thanks @jquense 🙌
Will you do a release with these changes? It would be awesome :)

**Pull request merged**